### PR TITLE
agent: reconnect after waiting 5 seconds

### LINKED ISSUE #8517: High load on management server due to stale KVM agent
##### ISSUE TYPE

 * Bug Report
 * Improvement Request
 * Other

##### COMPONENT NAME

~~~
KVM, Management Server
~~~

##### CLOUDSTACK VERSION

~~~
4.18.1, Any
~~~

##### CONFIGURATION

KVM 

##### OS / ENVIRONMENT

Rocky 8 KVM, Management server

##### SUMMARY

Removed KVM host with agent management server IP configured pointing to a new management server leads to high CPU load on the management server. 

Screenshot 2024-01-16 at 3 43 51 PM

We can see the below logs in the management server: 
```

2024-01-16 10:14:00,089 WARN  [c.c.u.n.Link] (AgentManager-SSLHandshakeHandler-381:null) (logid:) SSL Handshake has taken more than 30s to connect to: /10.1.35.66:46493. Please investigate this connection.
2024-01-16 10:14:00,099 WARN  [c.c.u.n.Link] (AgentManager-SSLHandshakeHandler-29:null) (logid:) SSL Handshake has taken more than 30s to connect to: /10.1.35.66:46475. Please investigate this connection.

```
##### STEPS TO REPRODUCE

~~~
1. Remove a KVM host. Restart the agent. 
2. Reinstall the management server or run cloudstack-setup-database
3. Watch the Management server CPU load. 

OR

1. Update the agent.properties of a KVM host to point it to an existing management server where the host is not added.
2. Restart the cloudstack agent.
3. Wait for a few minutes until the management server CPU load goes very high. 
~~~

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR fixes #8517 

see the steps to reproduce the issue in the description of #8517 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [x] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by weizhouapache** [comment]
@blueorangutan package

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by weizhouapache** [comment]
@blueorangutan test rocky8 kvm-rocky8

**Review by rohityadavcloud — APPROVED** [review]

**Comment by rohityadavcloud** [comment]
Could the mgmt server also sleep on such connections, on handshake exceptions? That is slow to respond such agent/clients?

**Comment by weizhouapache** [comment]
> Could the mgmt server also sleep on such connections, on handshake exceptions? That is slow to respond such agent/clients?

thanks @rohityadavcloud for the advice
I thought of it during my investigation. at the end I decided to change the agent as the connections are initialized by the agent. in my testing, the cpu load is lower than 10% I guess it is ok

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by rohityadavcloud** [comment]
@weizhouapache is this ready for review now?

**Comment by weizhouapache** [comment]
> @weizhouapache is this ready for review now?

yes @rohityadavcloud 
marked as ready for review

**Comment by DaanHoogland** [comment]
@blueorangutan LLtest alma9 kvm-alma9

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@blueorangutan test rocky8 kvm-rocky8

File: agent/src/main/java/com/cloud/agent/Agent.java [hunk 1]
```diff
@@ -1141,6 +1141,10 @@ public void doTask(final Task task) throws TaskExecutionException {
                     s_logger.error("Error parsing task", e);
                 }
             } else if (task.getType() == Task.Type.DISCONNECT) {
+                try {
```
Comment (vishesh92) [inline agent/src/main/java/com/cloud/agent/Agent.java:1144]: @weizhouapache Can we add some comment on why we have added a sleep here?

File: agent/src/main/java/com/cloud/agent/Agent.java [hunk 2]
```diff
@@ -1141,6 +1141,12 @@ public void doTask(final Task task) throws TaskExecutionException {
...
+                    // an issue has been found if reconnect inmediatelly after disconnecting. please refer to https://github.com/apache/cloudstack/issues/8517
```
Comment (sureshanaparti) [inline agent/src/main/java/com/cloud/agent/Agent.java:1145]:
```suggestion
                    // an issue has been found if reconnect immediately after disconnecting. please refer to https://github.com/apache/cloudstack/issues/8517
```

Comment (weizhouapache) [reply agent/src/main/java/com/cloud/agent/Agent.java]: thanks @sureshanaparti , committed

**Review by rajujith — APPROVED** [review]
LGTM.
Rogue KVM agent is unable to result high CPU load on the management server after this fix.

**Pull request merged**